roles/KubevirtMetricsAggregation/tasks/main.yml

```diff
@@ -6,4 +6,21 @@
     namespace: "{{ meta.namespace }}"
     definition: "{{ item | from_yaml }}"
   with_items: "{{ lookup('template', 'aggregation-rule-vmi-count.yaml.j2').split('\n---\n') | select('search', '(^|\n)[^#]') |list }}"
+  register: promrules
 
+- name: Inject ownerRef into the PrometheusRule definition
```

Comment:
This is not needed I think. You are not looking up existing PrometheusRoles and the ansible operator injects the proper reference automatically to newly created ones.

Reply:
The operator does not inject the reference if the object already exists, so I need to do it manually here in case the PrometheusRule already exists

Reply:
What about if it changes? This happens during upgrade so in theory the content might be different.

Reply:
I wonder if using `apply: yes` in the k8s task would fix that cleanly: https://docs.ansible.com/ansible/latest/modules/k8s_module.html#parameter-apply

It was introduced to Ansible 2.9 though, so you need to check what version the operator uses.

Reply:
According to this, owner references will never be automatically injected to existing objects:
https://github.com/operator-framework/operator-sdk/blob/master/website/content/en/docs/ansible/reference/retroactively-owned-resources.md

Reply:
So if it changes, the resource would be updated, but no new ownerRef would be set

Reply:
I see.